src/resource/resource.c

```diff
@@ -39,18 +39,26 @@ void recurse_dir(char *path) {
 			else
 				load_texture(buf);
 		} else
-#ifdef OGG_SUPPORT_ENABLED
-		if( (strcmp(dp->d_name + strlen(dp->d_name)-4, ".wav") == 0)
-		 || (strcmp(dp->d_name + strlen(dp->d_name)-4, ".ogg") == 0))
-#else
-		if  (strcmp(dp->d_name + strlen(dp->d_name)-4, ".wav") == 0)
-#endif
+		// A some kind of 'indian code', but...
+		if( (strcmp(dp->d_name + strlen(dp->d_name)-4, ".wav" ) == 0)
+		 || (strcmp(dp->d_name + strlen(dp->d_name)-4, ".ogg" ) == 0)
+		 || (strcmp(dp->d_name + strlen(dp->d_name)-4, ".mp3" ) == 0)
+		 || (strcmp(dp->d_name + strlen(dp->d_name)-4, ".mod" ) == 0)
+		 || (strcmp(dp->d_name + strlen(dp->d_name)-4, ".xm"  ) == 0)
+		 || (strcmp(dp->d_name + strlen(dp->d_name)-4, ".s3m" ) == 0)
+		 || (strcmp(dp->d_name + strlen(dp->d_name)-4, ".669" ) == 0)
+		 || (strcmp(dp->d_name + strlen(dp->d_name)-4, ".it"  ) == 0)
+		 || (strcmp(dp->d_name + strlen(dp->d_name)-4, ".med" ) == 0)
+		 || (strcmp(dp->d_name + strlen(dp->d_name)-4, ".mid" ) == 0)
+		 || (strcmp(dp->d_name + strlen(dp->d_name)-4, ".flac") == 0)
+		 || (strcmp(dp->d_name + strlen(dp->d_name)-4, ".aiff") == 0)
+		 || (strcmp(dp->d_name + strlen(dp->d_name)-4, ".voc" ) == 0))
```

Comment:
That’s fixed in my branch. This code also looks like it doesn’t compare 4-character extensions correctly.